### ISSUE TITLE
Global workqueue elements vanishing from central CouchDB

### ISSUE THREAD

**Issue by amaltaro** [dmwm/WMCore#9445]
**Impact of the bug**
BIG, impact still being assessed.

**Describe the bug**
Placeholder. But it looks like the cleanup global workqueue cherrypy thread hit an instability in central couchdb, and things got deleted..
From the workqueue logs, it seems it happened at this timestamp:
2019-11-28 18:02:21,285 (CERN time)
which would be around this in the couch.log
`[Thu, 28 Nov 2019 17:02:21 GMT]`

**How to reproduce it**

**Expected behavior**

**Additional context and error message**
Add any other context about the problem here, like error message and/or traceback. You might want to use triple back ticks to properly format it.

**Comment by amaltaro** [comment]
I still can't understand how this data got removed from the database, but one of the reasons we couldn't spot it in our monitoring is because the workqueue views seem to be stale, still reporting around 80k documents, while we have less than 20k currently in the database.
```
[.............]$ curl -ks -X GET 'http://localhost:5984/workqueue'
{"db_name":"workqueue","doc_count":18020,"doc_del_count":119910,"update_seq":256668,"purge_seq":0,"compact_running":false,"disk_size":711204987,"data_size":79897337,"instance_start_time":"1575052771233651","disk_format_version":6,"committed_update_seq":256668}
[.............]$ curl -ks -X GET 'http://localhost:5984/workqueue/_design/WorkQueue/_view/jobsByStatus?reduce=true&group=true'
{"rows":[
{"key":"Acquired","value":{"sum":234634,"count":4246,"min":1,"max":4132,"sumsqr":229624818}},
{"key":"Available","value":{"sum":4835103,"count":34223,"min":1,"max":3365,"sumsqr":4565244237}},
{"key":"Canceled","value":{"sum":1096,"count":2,"min":6,"max":1090,"sumsqr":1188136}},
{"key":"Done","value":{"sum":2063999,"count":7858,"min":1,"max":1762,"sumsqr":1991004713}},
{"key":"Running","value":{"sum":15106090,"count":39740,"min":1,"max":4620458,"sumsqr":23453900261130}}
]}
```

**Comment by vkuznet** [comment]
Alan,
one way to check this is to take FE logs and grep all couchdb calls, then
extract DN's to see who is doing calls against CouchDB.
Valentin.

**Comment by amaltaro** [comment]
That might be helpful, indeed. Scanning the couch logs and/or workqueue doesn't help because most of those operations are done in bulk (using the bulk couch api too). About WorkQueue, of course it does not provide a decent logging as well...

**Comment by amaltaro** [comment]
I had a look at one frontend and I don't see anything unusual at that time (around 28 Nov ~18:00:00). As I said before, there are a bunch of POST bulk updates to the workqueue from many agents.

Scanning the couch logs though (a shorter version of it is in vocms0740:/data/user/alan_couch.log), I see that we started compacting almost all databases (but wmstats AFAIK), thus heavily increasing the load on the node. A few seconds later there were issues reported in the log (but only for a few seconds), which made couch close a bunch of databases index, like:
```
[Thu, 28 Nov 2019 17:02:14 GMT] [info] [<0.19658.23>] Starting index update for db: workqueue idx: _design/WorkQueue
[Thu, 28 Nov 2019 17:02:14 GMT] [info] [<0.19658.23>] Index update finished for db: workqueue idx: _design/WorkQueue

[Thu, 28 Nov 2019 17:02:20 GMT] [info] [<0.3003.79>] Index shutdown by monitor notice for db: workqueue idx: _design/stupiddoc
[Thu, 28 Nov 2019 17:02:20 GMT] [info] [<0.3003.79>] Closing index for db: workqueue idx: _design/stupiddoc sig: "68728c971faed66e792daed86003047e"
[Thu, 28 Nov 2019 17:02:21 GMT] [info] [<0.19673.23>] Closing index for db: workqueue idx: _design/WorkQueue sig: "2f77345c5cf5065ec0f98a614222c8b2"
reason: {{timeout,
...
```

a few minutes later we got a CouchDB automatic restart:
```
[Thu, 28 Nov 2019 17:11:38 GMT] [info] [<0.31.0>] Apache CouchDB has started on http://0.0.0.0:5984/
```

and then I started seeing weird things in the logs, like (which is still there btw):
```
[Thu, 28 Nov 2019 23:40:01 GMT] [info] [<0.202.0>] Opening index for db: reqmgr_workload_cache idx: _design/ReqMgr sig: "c670a7f94b4f07f165eb7658067d1f69"
[Thu, 28 Nov 2019 23:40:01 GMT] [info] [<0.207.0>] Opening index for db: workqueue idx: _design/WorkQueue sig: "2f77345c5cf5065ec0f98a614222c8b2"
[Thu, 28 Nov 2019 23:40:01 GMT] [info] [<0.215.0>] Starting index update for db: workqueue idx: _design/WorkQueue
```

In addition to that, the 15min before 17:02:xx GMT had many many workqueue index update (almost 1hz updates..., very uncommon from scanning other logs):
```
Starting index update for db: workqueue idx: _design/WorkQueue
```

**Comment by amaltaro** [comment]
Checking the workqueue views at this very moment, I still see exactly the same values, even though the database is growing:
```
[.........]$ curl -ks -X GET 'http://localhost:5984/workqueue'
{"db_name":"workqueue","doc_count":31482,"doc_del_count":119910,"update_seq":323742,"purge_seq":0,"compact_running":false,"disk_size":1759342715,"data_size":105183574,"instance_start_time":"1575052771233651","disk_format_version":6,"committed_update_seq":323742}
[.........]$ curl -ks -X GET 'http://localhost:5984/workqueue/_design/WorkQueue/_view/jobsByStatus?reduce=true&group=true'
{"rows":[
{"key":"Acquired","value":{"sum":234634,"count":4246,"min":1,"max":4132,"sumsqr":229624818}},
{"key":"Available","value":{"sum":4835103,"count":34223,"min":1,"max":3365,"sumsqr":4565244237}},
{"key":"Canceled","value":{"sum":1096,"count":2,"min":6,"max":1090,"sumsqr":1188136}},
{"key":"Done","value":{"sum":2063999,"count":7858,"min":1,"max":1762,"sumsqr":1991004713}},
{"key":"Running","value":{"sum":15106090,"count":39740,"min":1,"max":4620458,"sumsqr":23453900261130}}
]}
[.........]$ date
Sun Dec  1 13:59:07 CET 2019
```
